workers/blobuploadcleanupworker/blobuploadcleanupworker.py

```diff
@@ -28,13 +28,31 @@ def __init__(self):
     def _try_cleanup_uploads(self):
         """
         Performs garbage collection on the blobupload table.
+        Will also perform garbage collection on the uploads folder in the S3 bucket,
+        if applicable.
         """
         try:
             with GlobalLock("BLOB_CLEANUP", lock_ttl=LOCK_TTL):
                 self._cleanup_uploads()
         except LockNotAcquiredException:
             logger.debug("Could not acquire global lock for blob upload cleanup worker")
-            return
+
+        if "FEATURE_CLEAN_UPLOADS_FOLDER" in app.config:
+            self._try_clean_partial_uploads()
```

Comment:
I think this should be in the same context as `BLOB_CLEANUP`, just to that it doesn't get preempted by another worker.

Reply:
Under the `with GlobalLock("BLOB_CLEANUP", lock_ttl=LOCK_TTL)`